services-js/registry-certs/client/birth/QuestionsFlow.tsx

```diff
@@ -10,116 +13,118 @@ import DateOfBirth from './questions/DateOfBirth';
+        ...answers,
+        forSelf,
+      });
+    } else {
+      this.props.birthCertificateRequest.answerQuestion(answers);
+    }
 
-  // When the user selects “back”, we treat that previous question
-  // as unanswered, and return to its screen.
-  private stepBackOneQuestion = () => {
-    const answeredQuestions = [...this.state.answeredQuestions];
-    const prevQuestion = answeredQuestions.pop()!;
-
-    const fieldsToClear: State = {} as any;
-
-    // If a question’s answer is made up of multiple fields,
-    // we must explicitly list them.
-    if (prevQuestion === 'forSelf') {
-      fieldsToClear.forSelf = null;
-      fieldsToClear.howRelated = '';
-    } else if (prevQuestion === 'bornInBoston') {
-      fieldsToClear.bornInBoston = '';
-      fieldsToClear.parentsLivedInBoston = '';
-    } else if (prevQuestion === 'nameOnRecord') {
-      fieldsToClear.firstName = '';
-      fieldsToClear.lastName = '';
-    } else if (prevQuestion === 'parentsNames') {
-      fieldsToClear.parent1FirstName = '';
-      fieldsToClear.parent1LastName = '';
-      fieldsToClear.parent2FirstName = '';
-      fieldsToClear.parent2LastName = '';
+    if (question === 'parentsNames') {
+      Router.push('/birth/review');
     } else {
-      fieldsToClear[prevQuestion] = '';
+      this.setState({ activeQuestion: this.state.questions[currentIndex + 1] });
     }
+  };
 
-    this.setState({
-      answeredQuestions,
-      activeQuestion: prevQuestion,
-      ...fieldsToClear,
-    });
+  private stepBackOneQuestion = (): void => {
+    const currentIndex = this.state.questions.indexOf(
+      this.state.activeQuestion
+    );
+    // Ensure we cannot go back any further than the first question.
+    const newIndex = currentIndex > 0 ? currentIndex - 1 : 0;
+
+    this.setState({ activeQuestion: this.state.questions[newIndex] });
   };
 
   // Clear all data and return to initial question.
   private handleUserReset = (): void => {
     this.setState({ ...INITIAL_STATE });
+    this.props.birthCertificateRequest.clearBirthCertificateRequest();
   };
 
-  private handleAnswers = (answers, currentQuestion, nextQuestion): void => {
-    this.setState({ ...answers });
-    this.proceedToNextQuestion(currentQuestion, nextQuestion);
-  };
+  public progressBar(): React.ReactChild {
+    const currentIndex = this.state.questions.indexOf(
+      this.state.activeQuestion
+    );
 
-  private handleForSelf = (answers): void => {
-    const forSelf = answers.forSelf === 'true';
+    return (
+      <progress
+        aria-label="Progress"
+        max="6"
+        value={currentIndex}
+        className={PROGRESS_BAR_STYLE}
+      >
+        Step {currentIndex + 1}
+      </progress>
+    );
+  }
 
-    this.setState({ ...answers, forSelf });
-    this.proceedToNextQuestion('forSelf', 'bornInBoston');
-  };
+  componentDidMount(): void {
+    // If parent1FirstName has a value, the user has come over from /review
```

Comment:
I think this logic belongs in the constructor, to set the initial state based on props. Otherwise you get an unnecessary re-render.

Reply:
Though I'm also realizing that if you kept `activeQuestion` in `birthCertificateRequest` rather than in this component’s state, you wouldn’t need this awkwardness.